lib/pbench/server/api/resources/compare_dataset.py

```diff
@@ -0,0 +1,145 @@
+            context: API context dictionary
+
+        Raises:
+            UnauthorizedAccess : The user isn't authorized for the requestedaccess.
+            APIAbort, reporting "NOT_FOUND" and "INTERNAL_SERVER_ERROR"
+            APIInternalError, reporting the failure message
+
+        GET /api/v1/compare?datasets=d1,d2,d3
+        """
+
+        datasets = params.query.get("datasets")
+        authorized_user: User = Auth.token_auth.current_user()
+
+        if not authorized_user:
+            raise APIAbort(
+                HTTPStatus.UNAUTHORIZED,
+                "User provided access_token is invalid or expired",
+            )
+
+        ds_list = [convert_dataset(ds, None) for ds in datasets]
+        cache_m = CacheManager(self.config, current_app.logger)
+
+        benchmark_check = []
+
+        stream_file = {}
+        for dataset in ds_list:
+            metadata = self._get_dataset_metadata(
+                dataset, ["dataset.metalog.pbench.script"]
+            )
+
+            benchmark = metadata["dataset.metalog.pbench.script"]
+            benchmark_check.append(benchmark)
+
+            # Validate if all the selected datasets is of same benchmark
+            if len(set(benchmark_check)) != 1:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    "Requested datasets are not of same benchmark. It can't be compared",
+                )
+
+            # Validate if the user is authorized to access the selected datasets
+            if (
+                dataset.owner_id != str(authorized_user.id)
+                and dataset.access == Dataset.PRIVATE_ACCESS
+            ):
+                raise UnauthorizedAccess(
+                    authorized_user,
+                    OperationCode.READ,
+                    dataset.owner.username,
+                    dataset.access,
+                    HTTPStatus.FORBIDDEN,
+                )
+
+        for dataset in ds_list:
+            try:
+                tarball = cache_m.find_dataset(dataset.resource_id)
+            except TarballNotFound as e:
+                raise APIAbort(
+                    HTTPStatus.NOT_FOUND, f"No dataset with ID '{e.tarball}' found"
+                ) from e
+            name = Dataset.stem(tarball.tarball_path)
+            try:
+                file = tarball.extract(tarball.tarball_path, f"{name}/result.csv")
+            except TarballUnpackError as e:
+                raise APIInternalError(str(e)) from e
+            stream_file[dataset.name] = file
+
+        benchmark = metadata["dataset.metalog.pbench.script"].upper()
+        benchmark_type = BenchmarkName.__members__.get(benchmark)
+        if not benchmark_type:
+            raise APIAbort(
+                HTTPStatus.UNSUPPORTED_MEDIA_TYPE, f"Unsupported Benchmark: {benchmark}"
+            )
+
+        get_quisby_data = QuisbyProcessing().compare_csv_to_json(
+            benchmark_type, InputType.STREAM, stream_file
+        )
+
```

Comment:
I realized during Riya's review that we should probably be careful that all consumers of the cache manager file streams should `stream_file.close()` after finishing to be sure that we don't accumulate too much IO buffer space before Python gets around to reclaiming dead objects.

Reply:
It's a wibbly wobbly timey wimey thing ... yeah, right now, for a fairly instantaneous (and therefore technically correct but globally limited) definition of "now", `Tarball.extract` returns a `str`; but I've been staring at Riya's PR which returns a stream, and the temporal dysphoria is real.

In fact, it's worth noting that the current `Tarball.extract` does a `tarfile.open` and then just lets the object go out of scope. Does `tarfile` have a close? Is it a context manager? Should/can `tarfile.extractfile` be used as a context manager, or closed? Hmm...

But that raises the question, if `stream_file` is actually a `dict[str,str]` where each value is the entire contents of a `.csv` file, should we be passing `InputType.STREAM` to Quisby? Or does this code, apparently much like me, think it's living in Riya's PR?